libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -0,0 +1,293 @@
+import { Moment } from 'moment';
+import { forkJoin } from 'rxjs';
+
+@Component({
+  selector: 'tamu-gisc-time-map',
+  templateUrl: './time-map.component.html',
+  styleUrls: ['./time-map.component.scss']
+})
+export class TimeMapComponent implements OnInit {
+  constructor(private mapService: MapboxMapService, private http: HttpClient) {}
+
+  private maxDate: string;
+  private mortalButtonToggled: boolean;
+  private stateButtonToggle: boolean;
+  private selectedDate: string;
+  private stateData: object;
+  private countyData: object;
+
+  public ngOnInit() {
+    const initObject = this;
+
+    this.mortalButtonToggled = false;
+    this.stateButtonToggle = false;
+    this.reloadData(this.mortalButtonToggled);
+
+    this.mapService.loaded.subscribe((map) => {
+      const zoomThreshold = 3;
+
+      map.addSource('county-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.countylines'
+      });
+
+      map.addSource('state-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.statelines'
+      });
+
+      map.addLayer(
+        {
+          id: 'covid-state',
+          type: 'fill',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          layout: {
+            // make layer visible by default (change to none or visible for debugging)
+            visibility: 'none'
+          }
+        },
+        'road-minor-low'
+      );
+
+      map.addLayer(
+        {
+          id: 'covid-county',
+          type: 'fill',
+          source: 'county-lines',
+          'source-layer': 'county-lines',
+          paint: {
+            'fill-outline-color': '#ffffff'
+          }
+        },
+        'covid-state'
+      );
+
+      map.addLayer(
+        {
+          id: 'statelines',
+          type: 'line',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          paint: {
+            'line-color': '#000000',
+            'line-width': 1
+          }
+        },
+        'state-label'
+      );
+
+      map.on('mousemove', 'covid-county', function(e) {
+        const feature = e.features[0];
+        const selectedCounty = initObject.countyData[initObject.selectedDate].filter(
+          (county) => county.fips === feature.properties.fips
+        );
+
+        document.getElementById('info-box').innerHTML =
+          feature.properties.NAME +
+          ' County' +
+          '</br>' +
+          'Population: ' +
+          feature.properties.POPESTIMATE2019 +
+          '</br>' +
+          'Cases: ' +
+          selectedCounty[0]['confirmed'] +
+          '</br>' +
+          'Infection Rate: ' +
+          selectedCounty[0]['infection_rate'].toFixed(2) +
+          '/100,000 People</br>' +
+          'Deaths: ' +
+          selectedCounty[0]['deaths'] +
+          '</br>' +
+          'Mortality Rate: ' +
+          selectedCounty[0]['death_rate'].toFixed(2) +
+          '/100,000 People';
+      });
```

Comment:
Separation of concerns. Do not compose templates in code, use actual templates.

Create a public member, update that, and use it in the template:


```js

public infoBoxModel;

this.infoBoxModel =  {
    county: feature.properties.Name,
    pop: feature.properties.POPESTIMATE2019,
    ...,
    ....
}
```

In template:

```html
    <p>County: {{infoBoxModel.county}}</p>
    <p>Population: {{infoBoxModel.pop}}</p>
```

Avoid the use of `<br>` tags for spacing.